### PR TITLE
Sprint 2 Final PR

### PR DESCRIPTION
**Pull request by Chaosking1623**
Added to grab_demo_testing.py and made very minor changes to other documents

### CODE REVIEW

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 1]
```diff
@@ -188,31 +203,36 @@ def CreateNewObjects(acConfigDlg, Restore):
...
+            DestroyObjects()
             print("m_Acquisition create failed")
+            return False
         else:
             print("m_Acquisition create success")
     
     if (m_Buffers != None and not m_Buffers.Initialized):
         if (m_Buffers.Create() == False):
-            #DestoryObjects()
+            DestroyObjects()
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py:291]: This isn't defined until later in the code. (You call `CreateNewObjects` on line 230, but don't define `DestroyObjects` until line 258). Please fix.

**Comment by bjones1** [comment]
This PR makes no sense -- it's a merge of something earlier, but doesn't have new code in it. What's going on?

**Comment by Chaosking1623** [comment]
I am not sure, should I close this  and then submit a new one after doing a pull? The files changed I believe shows some changes we made to grab_demo_testing_.py.

**Comment by bjones1** [comment]
Never mind -- my mistake. Just address review comments.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 2]
```diff
@@ -33,6 +33,7 @@
 #     intended to mimic how the grab demo initializes the components. The
 #     variable names and function names are the same in this file and within the
 #     grab demo.&nbsp;</p>
+from asyncio.windows_events import NULL
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Why is this imported? It's never used AFAIK.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 3]
```diff
@@ -163,23 +167,34 @@ def CreateNewObjects(acConfigDlg, Restore):
...
-        
+
 
         #m_Acquisition.SignalNotify += SapSignalNotifyHandler(grabDemo.GetSignalStatus)
         #m_Acquisition.SignalNotifyContext = grabDemo
     else:
         m_Buffers = SapBuffer()
         m_View = SapView(m_Buffers)
 
+    if (not(CreateObjects())):
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Should be `if not CreateObjects():`

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 4]
```diff
@@ -163,23 +167,34 @@ def CreateNewObjects(acConfigDlg, Restore):
...
+        DisposeObjects();
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Don't end lines with a semicolon -- this is Python, not C.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 5]
```diff
@@ -163,23 +167,34 @@ def CreateNewObjects(acConfigDlg, Restore):
...
     #     the grab demo due to function overloading. This is the start of the
     #     next function and where the errors occur.</p>
-    # <p>This section may need to be split into a separate function similar to
-    #     how it is within the grab demo to handle failure returns and so
+    # <p>CreateObjects() and CreateNewObjects() are separate, similar to how it
+    #     is within the grab demo to handle failure returns and so
     #     DisposeObjects() can be called at the proper time on failure.&nbsp;
     # </p>
+def CreateObjects():    
     if (m_Acquisition != None and not m_Acquisition.Initialized):
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Should be `if m_Acquisition is not None and not m_Acquisition.Initialized:`

See also line 212 and other similar items.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 6]
```diff
@@ -188,31 +203,36 @@ def CreateNewObjects(acConfigDlg, Restore):
...
             #     several other locations. Also needs to be called if any of the
             #     following creates fail as well.</p>
+            DestroyObjects()
             print("m_Acquisition create failed")
+            return False
         else:
             print("m_Acquisition create success")
     
     if (m_Buffers != None and not m_Buffers.Initialized):
         if (m_Buffers.Create() == False):
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Should be `if not m_Buffers.Create():`

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 7]
```diff
@@ -266,13 +287,88 @@ def DestroyObjects():
...
-#         style="background-color: rgb(248, 202, 198);">(TODO)</span>, is
-#     similar to DestroyObjects(). DisposeObjects() is supposed to be called
-#     when any of the objects fail to create and in the same locations where the
-#     connections are closed with DestroyObjects().&nbsp;</p>
+
+# <p>DisposeObjects() is supposed to be called when any of the objects fail to
+#     create and in the same locations where the connections are closed with
+#     DestroyObjects().</p>
+def DisposeObjects():
+    if (m_Xfer != None):
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: `if m_Xfer is not None:`, similarly for following lines.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 8]
```diff
@@ -266,13 +287,88 @@ def DestroyObjects():
...
+        m_Acquisition.Dispose()
+        m_Acquisition = None
+DisposeObjects()
+
+
+# <p>File Control</p>
+# <p>-----------------------------</p>
+# <p>Initialize() of the LoadSaveDlg.cs is needed ot get m_FileDialog.Filename
+# </p>
+if (m_isOpen):
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: `if m_isOpen:`

Also, this variable isn't declared yet...

**Review by bjones1 — CHANGES_REQUESTED** [review]

**Comment by Chaosking1623** [comment]
Added the suggested changes to fit Python syntax.

**Comment by bjones1** [comment]
Better, but note that Python doesn't need `if (condition):`, instead use `if condition:` (no extra parenthesis).

I still see an error when I run this:

```
C:> python grab_demo_testing.py
Cancel
False
0
Traceback (most recent call last):
  File "C:\Users\bjones\Documents\git\StandardMechanics\Image-acquisition-GUI\Python\grab_demo_testing.py", line 250, in <module>
    CreateNewObjects(acConfigDlg, False)
  File "C:\Users\bjones\Documents\git\StandardMechanics\Image-acquisition-GUI\Python\grab_demo_testing.py", line 178, in CreateNewObjects
    if (not(CreateObjects())):
  File "C:\Users\bjones\Documents\git\StandardMechanics\Image-acquisition-GUI\Python\grab_demo_testing.py", line 206, in CreateObjects
    DestroyObjects()
NameError: name 'DestroyObjects' is not defined
```

**Comment by Chaosking1623** [comment]
I made some small changes, but I am not sure if it will fix DestroyObjects() not being defined. I thought it would be defined as it is before CreateNewObjects is called.

**Comment by bjones1** [comment]
It looks like you're making progress! However, I still get syntax errors when I run the program -- please fix these and ensure that it runs.

```
(venv_std_mechanics_310) C:\Users\bjones\Documents\git\StandardMechanics\Image-acquisition-GUI\Python>python grab_demo_testing.py
  File "C:\Users\bjones\Documents\git\StandardMechanics\Image-acquisition-GUI\Python\grab_demo_testing.py", line 295
    print("m_Buffers create success"):
                                     ^
SyntaxError: invalid syntax
```

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 9]
```diff
@@ -163,56 +244,72 @@ def CreateNewObjects(acConfigDlg, Restore):
...
             print("m_View create failed")
+            return False
         else:
             print("m_View create success")
     
-    if (m_Xfer != None and not m_Xfer.Initialized):
-        if (m_Xfer.Create() == False):
-            #DestoryObjects()
+    if (m_Xfer is not None and not m_Xfer.Initialized):
+        if m_Xfer.Create() == False:
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: Use `if not m_Xfer.Create():` instead.

File: Image-acquisition-GUI/Python/grab_demo_testing.py [hunk 10]
```diff
@@ -234,45 +331,82 @@ def CreateNewObjects(acConfigDlg, Restore):
 #         EnableSignalStatus()</a>, is called from the first CreateNewObjects()
 #     function within the grab demo.</p>
 def EnableSignalStatus():
-    if (m_Acquisition != None):
+    if m_Acquisition != None:
```
Comment (bjones1) [inline Image-acquisition-GUI/Python/grab_demo_testing.py]: `if m_Acquisition is not None:`

**Review by bjones1 — CHANGES_REQUESTED** [review]
Still some syntax to update...

**Comment by CooperMedved** [comment]
Added the syntax changes that you requested
Is there anyway I can merge my changes to the file into the pr

**Comment by bjones1** [comment]
Yes, just make the changes in this (main) branch then push -- that will update the PR with your changes. Make sure the code runs without raising an exception!

**Comment by bjones1** [comment]
Should I review now, or do you still have more to push?

**Comment by bjones1** [comment]
Never mind -- I tracked down the bug (assignment to a global variable without using the `global` keyword). Good enough, thanks!

**Pull request merged**